src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Match.cs

```diff
@@ -87,7 +88,216 @@ public override bool CheckTypes(CheckTypesContext context, TexlNode[] args, DTyp
+                {
+                    var backRefName = groupMatch.Groups["goodBackRefName"].Value;
+
+                    // group isn't defined, or not defined yet
+                    if (!groupNameDict.TryGetValue(backRefName, out var groupNum))
+                    {
+                        errors.EnsureError(regExNode, TexlStrings.ErrInvalidRegExBadBackRefNotDefined, groupMatch.Value);
+                        return false;
+                    }
+
+                    // group is not closed and thus self referencing
+                    if (groupNumStack.Contains(groupNum))
+                    {
+                        errors.EnsureError(regExNode, TexlStrings.ErrInvalidRegExBadBackRefSelfReferencing, groupMatch.Value);
+                        return false;
+                    }
+                }
+                else if (groupMatch.Groups["badOctal"].Success)
+                {
+                    errors.EnsureError(regExNode, TexlStrings.ErrInvalidRegExBadOctal, groupMatch.Groups["badOctal"].Value);
+                    return false;
+                }
+                else if (groupMatch.Groups["badBalancing"].Success)
+                {
+                    errors.EnsureError(regExNode, TexlStrings.ErrInvalidRegExBadBalancing, groupMatch.Groups["badBalancing"].Value);
+                    return false;
+                }
+                else if (groupMatch.Groups["badOptions"].Success)
+                {
+                    if (groupMatch.Groups["badOptions"].Index > 0)
+                    {
+                        errors.EnsureError(regExNode, TexlStrings.ErrInvalidRegExBadOptionsNotAtFront, groupMatch.Groups["badOptions"].Value);
+                    }
+                    else
+                    {
+                        errors.EnsureError(regExNode, TexlStrings.ErrInvalidRegExBadOptions, groupMatch.Groups["badOptions"].Value);
+                    }
+
+                    return false;
+                }
+                else if (groupMatch.Groups["badSingleQuoteNamedCapture"].Success)
+                {
+                    errors.EnsureError(regExNode, TexlStrings.ErrInvalidRegExBadSingleQuoteNamedCapture, groupMatch.Groups["badSingleQuoteNamedCapture"].Value);
+                    return false;
+                }
+                else if (groupMatch.Groups["badConditional"].Success)
+                {
+                    errors.EnsureError(regExNode, TexlStrings.ErrInvalidRegExBadConditional, groupMatch.Groups["badConditional"].Value);
+                    return false;
+                }
+                else if (groupMatch.Groups["badEscapeAlpha"].Success)
+                {
+                    errors.EnsureError(regExNode, TexlStrings.ErrInvalidRegExBadEscape, groupMatch.Groups["badEscapeAlpha"].Value);
+                    return false;
+                }
+                else
+                {
+                    throw new NotImplementedException("Unknown regular expression match");
```

Comment:
Should this ever be hit?

Reply:
No, this should never be hit.  But is a good block to have if the names in the RE and the long if() are off.  Added a comment.